File: src/windows/Maple.WinUI/MapleUI/Templates/MuiOverlayShellMath.cs
namespace Maple.UI
{
    /// <summary>Overlay Shell panel size (unified-component-catalog.md
    /// §5, "Overlay Shell" row: "sm, md, lg, full"). Declared here (not in
    /// the WinUI-dependent <see cref="MuiOverlayShell"/> control file) so
    /// this WinUI-free enum + its size-mapping math can be linked into
    /// Maple.WinUI.Tests without pulling in WinUI — same split as
    /// <see cref="MuiDialogResultKind"/>/<see cref="MuiDialogLogic"/>.
    /// </summary>
    public enum MuiOverlayShellSize { Sm, Md, Lg, Full }

    /// <summary>
    /// Pure size → pixel-width mapping behind <see cref="MuiOverlayShell"/>
    /// — unit-tested without a live Window. Mirrors
    /// `mui-overlay-shell.component.scss`'s <c>.size-*</c> max-widths.
    /// </summary>
    public static class MuiOverlayShellMath
    {
        public static double MaxWidthFor(MuiOverlayShellSize size) => size switch
        {
            MuiOverlayShellSize.Sm => 360.0,
            MuiOverlayShellSize.Lg => 800.0,
            MuiOverlayShellSize.Full => double.PositiveInfinity,
            _ => 560.0, // Md, and the safe fallback for any future value.
        };
    }
}

File: src/apple/Packages/MapleUI/Sources/MapleUI/Internal/MuiOverlayShellMath.swift
// MuiOverlayShellMath.swift — pure size→max-width mapping for
// MuiOverlayShell (Templates, unified-component-catalog.md §5). Matches the
// web reference's `mui-overlay-shell.component.scss` size classes
// (`.size-sm` / `.size-md` / `.size-lg` / `.size-full`) so the two
// platforms present visually consistent modal widths.

import Foundation

enum MuiOverlayShellMath {
    /// The panel's max width in points for `size`, or `nil` for `.full`
    /// (no cap — the panel fills the available width).
    static func maxWidth(for size: MuiOverlayShellSize) -> Double? {
        switch size {
        case .sm: return 360
        case .md: return 560
        case .lg: return 800
        case .full: return nil
        }
    }
}
